### PR TITLE
Progressbar inside treeview

### PR DESCRIPTION
**Pull request by actionless**
the problem can be noticed for example in transmission-gtk

### CODE REVIEW

**Comment by megax** [comment]
Code fixed/changed and fixed gnome-system-monitor:

```
        .progressbar {
            @include linear-gradient($selected_bg_color);
            border: 1px solid $base_color;

            &:selected {
                &:focus, & {
                    @include linear-gradient($selected_bg_color);
                    border: 1px solid mix($selected_bg_color, $selected_fg_color, .2);
                    color: $selected_fg_color;
                }

                &:backdrop { }
            }

            &:backdrop { }
        }

        .trough {
            background-color: mix($bg_color, $base_color, .2);
            border: 1px solid $base_color;

            &:selected {
                &:focus, & {
                    background-color: mix($bg_color, $base_color, .5);
                    border: 1px solid $bg_color;
                    color: $text_color;
                }
            }
        }
```

**Comment by satya164** [comment]
Can you fix the lint warnings and give before/after screenshots?

**Comment by actionless** [comment]
@megax removing `:insensitive` and `:insensitive:insensitive` in your patch breaks progressbar for paused torrent

**Comment by actionless** [comment]
@satya164 
2016-03-26--1459002136_1492x2062_scrot
2016-03-26--1459002213_1492x2062_scrot

**Comment by megax** [comment]
gnome-system-monitor: before: https://www.dropbox.com/s/lz9urgamjzujelm/2016.03.26.-15.24.14.png?dl=0 after: https://db.tt/bG8jxo5x

**Comment by actionless** [comment]
@megax i can't open the first link

**Comment by megax** [comment]
Fixed :)

**Comment by actionless** [comment]
@megax now both appear to be the same

**Comment by actionless** [comment]
gnome-system-monitor

before:
2016-03-26--1459002609_1874x904_scrot

after:
2016-03-26--1459002601_1874x904_scrot

**Comment by satya164** [comment]
Can you fix the lint warnings so we can merge?

**Comment by megax** [comment]
Focus color not ideal (white font and background: white): https://db.tt/VjrzdPUQ

**Comment by actionless** [comment]
@megax i think it can be fixed with styling label fg inside .progressbar;selected:focus

**Comment by actionless** [comment]
@satya164 commited

**Comment by satya164** [comment]
Can you also squash the commits to one, please?

**Comment by actionless** [comment]
yup, sorry, i was backporting that change from my fork (https://github.com/actionless/oomox) so was just cherry-picking the relevant commits

**Comment by actionless** [comment]
squashed

File: gtk-3.0/scss/widgets/_view.scss [hunk 1]
```diff
@@ -126,6 +126,27 @@
...
+        .trough:selected,
+        .trough:selected:focus {
+            background-color: mix($bg_color, $base_color, 0.5);
+            border: 1px solid $bg_color;
+        }
+        .progressbar:insensitive,
+        .progressbar:insensitive:insensitive {
+            @include linear-gradient($bg_color);
+            border-color: $bg_color;
+        }
```
Comment (satya164) [inline gtk-3.0/scss/widgets/_view.scss:148]: You left one lint warning.

**Comment by megax** [comment]
My code transmission: https://db.tt/iFwhHcya

**Comment by actionless** [comment]
weird, when i was trying it it was completely white

**Comment by megax** [comment]
@actionless Combined and fixed color (border) and converted sass code format:

> ```
>     .progressbar {
>         @include linear-gradient($selected_bg_color);
>         border: 1px solid border_normal($selected_bg_color);
> 
>         &:selected {
>             &:focus, & {
>                 @include linear-gradient($selected_bg_color);
>                 border: 1px solid border_focus($selected_bg_color);
>                 color: $selected_fg_color;
>             }
> 
>             &:backdrop { }
>         }
> 
>         &:insensitive {
>             &:insensitive {
>                 @include linear-gradient($bg_color);
>                 border-color: border_insensitive($bg_color);
>             }
>         }
> 
>         &:backdrop {
>             @include linear-gradient($bg_color);
>             border-color: border_insensitive($bg_color);
>         }
>     }
> 
>     .trough {
>         background-color: mix($bg_color, $base_color, .5);
>         border: 1px solid border_normal($base_color);
> 
>         &:selected {
>             &:focus, & {
>                 background-color: $base_color;
>                 border: 1px solid border_focus($bg_color);
>                 color: $text_color;
>             }
>         }
> 
>         &:insensitive {
>             &:insensitive {
>                 background-color: $base_color;
>             }
>         }
>     }
> ```

Sample:
gnome-system-monitor: https://db.tt/8SFAoGhF
transmission: insensitive: selected https://db.tt/BrzaOCbE and not selected: https://db.tt/WveZjFZ1
transmission: normal: https://db.tt/vXXqL0h1

transmission and gnome-system-monitor: backdrop: https://db.tt/E278bMQD

**Comment by satya164** [comment]
Merging this. @megax Would be awesome if you send a followup PR.

**Pull request merged**